### ISSUE TITLE
`import/no-duplicates` should take import attributes into consideration

### ISSUE THREAD

**Issue by maxmilton** [import-js/eslint-plugin-import#3290]
Inside some of my unit tests I have import code (using [Bun macro import annotation](https://bun.sh/docs/bundler/macros#import-attributes)) like:

```ts
import { compile } from "../src/macro.ts" with { type: "macro" };
import { compile as compileNoMacro } from "../src/macro.ts";
```

This currently reports an error, even though the [import attributes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with) are different. It shouldn't be restricted to just the `"type"` attribute key, any differing import attributes should be treated as a unique import even with the same import path.

I'd like to see a new option added to `import/no-duplicates` (and enabled by default) to differentiate import by their import attributes.

**Comment by ljharb** [comment]
I don't think an option makes sense; I think this is just a bug/oversight. (that said, I strongly discourage using any attributes that aren't part of an actual standard).

**Comment by maxmilton** [comment]
Actually yes, I agree it shouldn't be an option but default behaviour. I can't think of a scenario where I would want to disable it.

The only standard import attributes I could find are:

```ts
// ECMAScript standard
{ type: "json" | "text" } // also "bytes" is at proposal stage

// HTML spec
{ type: "json" | "text" | "css" }
```

Popular runtimes:

```ts
// TypeScript
{ "resolution-mode": "import" | "require" }

// Node.js
// Arbitrary attributes visible to custom loaders
// Same as ECMAScript standard by default

// Bun
{ type: "js" | "jsx" | "ts" | "tsx" | "json" | "jsonc" | "json5" | "toml" | "yaml" | "xml" | "text" | "bytes" | "napi" | "sqlite" | "html" | "css" | "file" }
{ type: "macro" }
{ type: "sqlite", embed: "true" }

// Deno
{ type: "json" | "text" | "bytes" | "css" }
```

Popular build tools:

```ts
// esbuild
{ type: "json" | "text" | "bytes" }

// Webpack 5
// Handles arbitrary attributes

// Rollup
// Handles arbitrary attributes

// Vite
// Not supported; at proposal stage
```

Looking over all that, at the very least the "type" attribute should make the import unique. But given that Node, Webpack, and Rollup currently support arbitrary import attributes, this rule probably should too.
